src/backend/resourcemanager/communication/rmcomm_QD2RM.c

```diff
@@ -1438,6 +1438,7 @@ void freeHeartBeatThreadArg(HeartBeatThreadArg *arg)
 	{
 		free((*arg)->HostAddrs[i]);
 	}
+	free((*arg)->HostAddrs);
```

Comment:
have a look at the code: tharg->HostAddrs = malloc(sizeof(char *) * tharg->HostAddrSize);
where tharg->HostAddrSize maybe zero and malloc(0) happens.
what i mean yesterday is to test whether tharg->HostAddrSize > 0 before free arg->HostAddrs